pkg/util/rerun/rerun.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rerun
+
+import (
+	"github.com/golang/glog"
+)
+
+// ReRun will restart the target function when event gets changed
```

Comment:
This comment is insufficient to describe the complexity that this is trying to capture.  I think this function is kind of over-complicated.  Can you write a nice comment explaining what you want to do, what all the args are for, and then see if the code can be simpler?